torch/onnx/symbolic_opset11.py

```diff
@@ -117,3 +117,37 @@ def unique_dim(g, self, dim, sorted, return_inverse, return_counts):
 
 def round(g, self):
     return g.op("Round", self)
+
+
+def arange(g, *args):
+    def _get_arange_dtype(dtype):
+        dtype = sym_help._maybe_get_const(dtype, 'i')
+        if sym_help._is_value(dtype):
+            dtype = 4  # default to int64
+        return dtype
+
+    if len(args) == 5:
+        # aten::arange(Scalar end, ScalarType dtype, Layout, Device, bool pin_memory)
+        dtype = _get_arange_dtype(args[1])
+        start_default = g.op("Constant", value_t=torch.tensor(0, dtype=sym_help.scalar_type_to_pytorch_type[dtype]))
+        delta_default = g.op("Constant", value_t=torch.tensor(1, dtype=sym_help.scalar_type_to_pytorch_type[dtype]))
+        arange_tensor = g.op("Range", start_default, args[0], delta_default)
```

Comment:
I am looking at the onnx doc for range and noticed that start/limit/delta should be of type T, and output is of type T.

would that mean that if dtype is float for example, and args[0] is an int it will fail?
if so, should we add a cast(dtype) on args[0]?

Reply:
I added Range to the list in scalar_type_analysis pass. I think that will fix this for Range op.
For example this test: test_dyn_arange passed

Reply:
I am not convinced..

for example
torch.arange(0,3,1, dtype=torch.float)

will go in the elif len(args)==7, and create g.op("Range", args[0], args[1], args[2]).
But args[0], args[1], args[2] are all ints so wouldn't the output be an int?

Reply:
oh yes I see this. For the last case len(args)==7 I'd need a cast.

Reply:
I think other cases should be covered. I'll add a test to check that.

Reply:
I just updated for this fix.